File: java/1332.删除回文子序列.java
class Solution {
    // 1. O(n) t:0ms O(1) m:36.2M(53%)
    public int removePalindromeSub(String s) {
        int n = s.length();
        for (int i = 0; i <= n / 2; i++){
            if (s.charAt(i) != s.charAt(n - i - 1)) return 2;
        }
        return 1;
    }
}

File: python/1332.删除回文子序列.py
class Solution:
    # 1. O(n) t:28ms(85%) O(n) m:15.1M(10%)
    def removePalindromeSub(self, s: str) -> int:
        return 1 if s == s[::-1] else 2
    
    # # 2. O(n) t:32ms(65%) O(1) m:15.1M(6%)
    # def removePalindromeSub(self, s: str) -> int:
    #     for i in range(len(s)//2):
    #         if s[i] != s[len(s)-1-i]:
    #             return 2
    #     return 1